tests/api_fastapi/views/public/test_dag_run.py

```diff
@@ -0,0 +1,130 @@
+DAG2_ID = "test_dag2"
+DAG1_RUN1_ID = "dag_run_1"
+DAG1_RUN2_ID = "dag_run_2"
+DAG2_RUN1_ID = "dag_run_3"
+DAG2_RUN2_ID = "dag_run_4"
+DAG1_RUN1_STATE = DagRunState.SUCCESS
+DAG1_RUN2_STATE = DagRunState.FAILED
+DAG2_RUN1_STATE = DagRunState.SUCCESS
+DAG2_RUN2_STATE = DagRunState.SUCCESS
+DAG1_RUN1_RUN_TYPE = DagRunType.MANUAL
+DAG1_RUN2_RUN_TYPE = DagRunType.SCHEDULED
+DAG2_RUN1_RUN_TYPE = DagRunType.BACKFILL_JOB
+DAG2_RUN2_RUN_TYPE = DagRunType.DATASET_TRIGGERED
+DAG1_RUN1_TRIGGERED_BY = DagRunTriggeredByType.UI
+DAG1_RUN2_TRIGGERED_BY = DagRunTriggeredByType.DATASET
+DAG2_RUN1_TRIGGERED_BY = DagRunTriggeredByType.CLI
+DAG2_RUN2_TRIGGERED_BY = DagRunTriggeredByType.REST_API
+START_DATE = datetime(2024, 6, 15, 0, 0, tzinfo=timezone.utc)
+EXECUTION_DATE = datetime(2024, 6, 16, 0, 0, tzinfo=timezone.utc)
+DAG1_NOTE = "test_note"
+
+
+@pytest.fixture(autouse=True)
+@provide_session
+def setup(dag_maker, session=None):
+    clear_db_runs()
+    clear_db_dags()
+    clear_db_serialized_dags()
+
+    with dag_maker(
+        DAG1_ID,
+        schedule="@daily",
+        start_date=START_DATE,
+    ):
+        EmptyOperator(task_id="task_1")
+    dag1 = dag_maker.create_dagrun(
+        run_id=DAG1_RUN1_ID,
+        state=DAG1_RUN1_STATE,
+        run_type=DAG1_RUN1_RUN_TYPE,
+        triggered_by=DAG1_RUN1_TRIGGERED_BY,
+    )
+    dag1.note = (DAG1_NOTE, 1)
+
+    dag_maker.create_dagrun(
+        run_id=DAG1_RUN2_ID,
+        state=DAG1_RUN2_STATE,
+        run_type=DAG1_RUN2_RUN_TYPE,
+        triggered_by=DAG1_RUN2_TRIGGERED_BY,
+        execution_date=EXECUTION_DATE,
+    )
+
+    with dag_maker(
+        DAG2_ID,
+        schedule=None,
+        start_date=START_DATE,
+    ):
+        EmptyOperator(task_id="task_2")
+    dag_maker.create_dagrun(
+        run_id=DAG2_RUN1_ID,
+        state=DAG2_RUN1_STATE,
+        run_type=DAG2_RUN1_RUN_TYPE,
+        triggered_by=DAG2_RUN1_TRIGGERED_BY,
+        execution_date=EXECUTION_DATE,
+    )
+    dag_maker.create_dagrun(
+        run_id=DAG2_RUN2_ID,
+        state=DAG2_RUN2_STATE,
+        run_type=DAG2_RUN2_RUN_TYPE,
+        triggered_by=DAG2_RUN2_TRIGGERED_BY,
+        execution_date=EXECUTION_DATE,
+    )
+
+    dag_maker.dagbag.sync_to_db()
+    dag_maker.dag_model
+    dag_maker.dag_model.has_task_concurrency_limits = True
+    session.merge(dag_maker.dag_model)
+    session.commit()
+
+
+@pytest.mark.parametrize(
+    "dag_id, run_id, state, run_type, triggered_by, dag_run_note",
+    [
+        (DAG1_ID, DAG1_RUN1_ID, DAG1_RUN1_STATE, DAG1_RUN1_RUN_TYPE, DAG1_RUN1_TRIGGERED_BY, DAG1_NOTE),
+        (DAG1_ID, DAG1_RUN2_ID, DAG1_RUN2_STATE, DAG1_RUN2_RUN_TYPE, DAG1_RUN2_TRIGGERED_BY, None),
+        (DAG2_ID, DAG2_RUN1_ID, DAG2_RUN1_STATE, DAG2_RUN1_RUN_TYPE, DAG2_RUN1_TRIGGERED_BY, None),
+        (DAG2_ID, DAG2_RUN2_ID, DAG2_RUN2_STATE, DAG2_RUN2_RUN_TYPE, DAG2_RUN2_TRIGGERED_BY, None),
+    ],
+)
+def test_get_dag_run(test_client, dag_id, run_id, state, run_type, triggered_by, dag_run_note):
```

Comment:
Maybe opt for `class oriented` tests. For instance you can take a look to `connections` tests.

I will update `dags`, because this is more consistant with `class oriented` airflow tests. (most of them are written like this).

Also you can add a test case for 404.

Reply:
@pierrejeambrun , Do you want me to update tests to Class based on this PR or another one?

Reply:
I would like to work on updating tests to class-based tests for both tests_dag.py and tests_dag_run.py in a separate PR. Hope that's fine?